src/iris/app_stats.py

```diff
@@ -84,116 +98,180 @@ def calculate_app_stats(app, connection, cursor, fields_filter=None):
+                stats['pct_%s_fail_last_week' % mode][formatted_date] = fail_pct
+                stats['pct_%s_other_last_week' % mode][formatted_date] = other_pct
+            else:
+                stats['pct_%s_success_last_week' % mode] = {}
+                stats['pct_%s_fail_last_week' % mode] = {}
+                stats['pct_%s_other_last_week' % mode] = {}
+
+                stats['pct_%s_success_last_week' % mode][formatted_date] = success_pct
+                stats['pct_%s_fail_last_week' % mode][formatted_date] = fail_pct
+                stats['pct_%s_other_last_week' % mode][formatted_date] = other_pct
+
+        # Get counts of messages sent per mode
+        cursor.execute('''SELECT `mode`.`name`, `msg_count` FROM
+                            (SELECT `mode_id`, COUNT(*) AS `msg_count` FROM `message`
+                            USE INDEX (ix_message_sent)
+                            WHERE `sent` > %(date_variable)s - INTERVAL 1 WEEK
+                            AND `sent` < %(date_variable)s
+                            AND `application_id` = %(application_id)s
+                            GROUP BY `mode_id`) `counts`
+                        JOIN `mode` ON `mode`.`id` = `counts`.`mode_id`''',
+                       query_data)
+
+        for row in cursor:
+            key = 'total_%s_sent_last_week' % row[0]
+            result = row[1]
+
+            if stats.get(key):
+                stats[key][formatted_date] = result
+            else:
+                stats[key] = {}
+                stats[key][formatted_date] = result
+
+        # Zero out modes that don't show up in the count
+        for mode in modes:
+            count_stat = 'total_%s_sent_last_week' % mode
+            if count_stat not in stats:
+                stats[count_stat] = {}
+                stats[count_stat][formatted_date] = 0
+            elif formatted_date not in stats[count_stat]:
+                stats[count_stat][formatted_date] = 0
+
+    for stat, nested_dict in stats.items():
+        for timestamp, value in nested_dict.items():
+            if value is None:
+                stats[stat][timestamp] = -1
+
+    return stats
+
+
+def calculate_high_priority_incidents(connection, cursor):
+
+    high_priority_query = '''SELECT application.name, COUNT(DISTINCT incident.application_id, incident.id) AS incident_count FROM incident JOIN application ON application.id = incident.application_id JOIN message ON message.incident_id = incident.id WHERE incident.created > %(date_variable)s - INTERVAL 1 WEEK AND incident.created < %(date_variable)s AND priority_id IN ( SELECT id FROM priority WHERE name IN ('high', 'urgent') ) GROUP BY incident.application_id ORDER BY incident_count DESC'''
+
+    stats = {}
+
+    for delta in range(0, 7):
+        now = datetime.utcnow() - timedelta(weeks=delta)
+        formatted_date = now.strftime('%Y-%m-%d %H:%M:%S')
+        stats[formatted_date] = []
```

Comment:
When we're passing datetimes around, we should always use Unix timestamps to avoid timezone shenanigans. Converting to a string may/may not be necessary when talking to the DB, you'll need to double check that. For the stats variable, we should keep it in Unix time, then convert to pretty print in the frontend if needed.